tiledb/sm/filesystem/posix.cc

```diff
@@ -70,43 +70,23 @@ bool Posix::both_slashes(char a, char b) {
 }
 
 uint64_t Posix::read_all(
-    int fd, void* buffer, uint64_t nbytes, uint64_t offset) {
+    int fd, void* buffer, uint64_t nbytes, uint64_t offset, bool& failed) {
+  failed = false;
   auto bytes = reinterpret_cast<char*>(buffer);
   uint64_t nread = 0;
   do {
     ssize_t actual_read =
         ::pread(fd, bytes + nread, nbytes - nread, offset + nread);
-    if (actual_read == -1) {
-      LOG_STATUS_NO_RETURN_VALUE(
-          Status_Error(std::string("POSIX pread error: ") + strerror(errno)));
-      return nread;
-    } else {
-      nread += actual_read;
+    if (actual_read <= 0) {
```

Comment:
I'm not a big fan of using the failed flag here as its mixing the logic quite a bit. What if we change the return value of read_all to Status, then we can just do:

```c++
if(actual_read < 0) {
  return LOG_STATUS(Status_IOError("POSIX read error: ") + strerror(errno));
} else if(actual_read == 0) {
  return LOG_STATUS(Status_IOError("POSIX incomplete read: EOF reached"));
} else {
  nread += actual_read;
}
```

And then below in read we can just do:

```c++
auto st = read_all(fd, buffer, nbytes, offset);
if(!st.ok()) {
  close(fd);
  return st;
}
```

Reply:
OK, the `Status` will be removed in a future refactoring, once both VFS PRs get merged.